modules/view/test/spec.coffee

```diff
@@ -1,19 +1,50 @@
 describe 'hx-view', ->
+  origConsoleWarning = hx.consoleWarning
+
+  beforeEach ->
+    hx.consoleWarning = chai.spy()
+
+  afterEach ->
+    hx.consoleWarning = origConsoleWarning
+
   it "should create elements of a given type using the default view enter fn", ->
     selection = hx.detached('div')
     selection.view('element-type').apply([1..3])
     selection.selectAll('element-type').size().should.equal(3)
+    hx.consoleWarning.should.not.have.been.called()
```

Comment:
Added this to every function because it shouldn't be called in these situations. We are also now testing for cases when it _should_ be called.